File: src/dotnet/Core/Domain/Configuration/RankingWeights.cs
namespace Semantic.Kernel.Api.Core.Domain.Configuration;

public sealed record RankingWeights
{
    public required double TechnicalMatchWeight { get; init; }
    public required double SeniorityMatchWeight { get; init; }
    public required double LeadershipSignalsWeight { get; init; }
    public required double ExperienceMatchWeight { get; init; }
    public required int LeadershipKeywordThreshold { get; init; }
    public required int MaxSeniorityDelta { get; init; }
    public required double MaxLeadershipContribution { get; init; }
    
    public static RankingWeights Default() => new()
    {
        TechnicalMatchWeight = 0.40,
        SeniorityMatchWeight = 0.25,
        LeadershipSignalsWeight = 0.20,
        ExperienceMatchWeight = 0.15,
        LeadershipKeywordThreshold = 2,
        MaxSeniorityDelta = 2,
        MaxLeadershipContribution = 0.5
    };
}

File: src/python/core/domain/configuration/ranking_weights.py
from dataclasses import dataclass


@dataclass(frozen=True)
class RankingWeights:
    technical_match_weight: float
    seniority_match_weight: float
    leadership_signals_weight: float
    experience_match_weight: float
    leadership_keyword_threshold: int
    max_seniority_delta: int
    max_leadership_contribution: float
    
    @staticmethod
    def default() -> "RankingWeights":
        return RankingWeights(
            technical_match_weight=0.40,
            seniority_match_weight=0.25,
            leadership_signals_weight=0.20,
            experience_match_weight=0.15,
            leadership_keyword_threshold=2,
            max_seniority_delta=2,
            max_leadership_contribution=0.5
        )
